src/work_handler.ts

```diff
@@ -72,34 +77,61 @@ export class WorkHandler {
    */
   async processEvent(appEvent: AppEvent): Promise<void> {
     switch (appEvent.type) {
+      case EventType.VALIDATE_POOLS: {
+        for (const config of appEvent.pools) {
+          try {
+            let pool = await this.sorobanHelper.loadPool(config);
+            if (pool.metadata.backstop !== APP_CONFIG.backstopAddress) {
+              throw new Error(
+                `Backstop address for pool: ${config.poolAddress} is not the expected address: ${APP_CONFIG.backstopAddress}`
+              );
+            }
+          } catch (error) {
+            throw new Error(
+              `Failed to load pool: ${config.poolAddress} please check that the pool config is correct. Error: ${error}`
+            );
+          }
+        }
+        break;
+      }
+
       case EventType.PRICE_UPDATE: {
         await setPrices(this.db);
         break;
       }
       case EventType.ORACLE_SCAN: {
-        const poolOracle = await this.sorobanHelper.loadPoolOracle();
-        const priceChanges = this.oracleHistory.getSignificantPriceChanges(poolOracle);
-        // @dev: Insert into a set to ensure uniqueness
-        let usersToCheck = new Set<string>();
-        for (const assetId of priceChanges.up) {
-          const usersWithLiability = this.db.getUserEntriesWithLiability(assetId);
-          for (const user of usersWithLiability) {
-            usersToCheck.add(user.user_id);
+        for (const poolConfig of APP_CONFIG.poolConfigs) {
+          let usersToCheck = new Set<string>();
+          const poolOracle = await this.sorobanHelper.loadPoolOracle(poolConfig);
+          const priceChanges = this.oracleHistory.getSignificantPriceChanges(poolOracle);
+          // @dev: Insert into a set to ensure uniqueness
```

Comment:
oracle prices also need to be separated by pool, as they could be using different oracles entirely.